k9mail/src/main/java/com/fsck/k9/crypto/CryptoHelper.java

```
@@ -22,22 +23,78 @@
                     ".*?(-----BEGIN PGP SIGNED MESSAGE-----.*?-----BEGIN PGP SIGNATURE-----.*?-----END PGP SIGNATURE-----).*",
                     Pattern.DOTALL);
 
+    public static final String APPLICATION_PKCS7_MIME = "application/pkcs7-mime";
+    public static final String MULTIPART_ENCRYPTED = "multipart/encrypted";
+    public static final String MULTIPART_SIGNED = "multipart/signed";
+    public static final String PROTOCOL_PARAMETER = "protocol";
+    public static final String APPLICATION_PGP_ENCRYPTED = "application/pgp-encrypted";
+    public static final String APPLICATION_PGP_SIGNATURE = "application/pgp-signature";
+    public static final String TEXT_PLAIN = "text/plain";
+
     public CryptoHelper() {
         super();
     }
 
     /**
-     * TODO: use new parseMessage() from PgpUtils to actually parse!
      * @param message
      * @return
      */
-    public boolean isEncrypted(Message message) {
+    public static boolean isSMimeEncrypted(Message message) {
+        String[] contentHeaders = null;
+        try {
+            contentHeaders = message.getHeader("Content-Type");
+        } catch (MessagingException e) {
+            // guess not...
```

please rethrow rather than swallowing